File: app/src/main/java/com/kotyj/com/activity/mine/WithdrawListActivity.java
package com.kotyj.com.activity.mine;

import android.os.Bundle;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.OrientationHelper;
import android.support.v7.widget.RecyclerView;
import android.widget.ImageView;
import android.widget.TextView;

import com.alibaba.fastjson.JSONArray;
import com.kotyj.com.R;
import com.kotyj.com.adapter.WithdrawListAdapter;
import com.kotyj.com.base.BaseActivity;
import com.kotyj.com.model.BaseEntity;
import com.kotyj.com.model.WithdrawModel;
import com.kotyj.com.utils.ViewUtils;
import com.kotyj.com.utils.okgo.OkClient;
import com.lzy.okgo.model.HttpParams;
import com.lzy.okgo.model.Response;
import com.scwang.smartrefresh.layout.SmartRefreshLayout;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class WithdrawListActivity extends BaseActivity {


    @BindView(R.id.iv_back)
    ImageView ivBack;
    @BindView(R.id.tv_title)
    TextView tvTitle;
    @BindView(R.id.tv_right)
    TextView tvRight;
    @BindView(R.id.iv_right)
    ImageView ivRight;
    @BindView(R.id.recyclerView)
    RecyclerView rvList;
    @BindView(R.id.smartRefreshLayout)
    SmartRefreshLayout smartRefreshLayout;
    private WithdrawListAdapter mWithdrawListAdapter;
    private List<WithdrawModel> mList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // TODO: add setContentView(...) invocation
        ButterKnife.bind(this);
    }

    @Override
    public int initLayout() {
        return R.layout.general_mlist;
    }

    @Override
    public void initData() {
        tvTitle.setText("提现记录");
        smartRefreshLayout.setEnableLoadmore(false);
        smartRefreshLayout.setEnableRefresh(false);
        mWithdrawListAdapter = new WithdrawListAdapter(mList);
        rvList.setLayoutManager(new LinearLayoutManager(context));
        rvList.addItemDecoration(new DividerItemDecoration(context, OrientationHelper.VERTICAL));
        mWithdrawListAdapter.bindToRecyclerView(rvList);
        mWithdrawListAdapter.setEmptyView(R.layout.layout_bank_empty, rvList);
        loadData();
    }

    private void loadData() {
        loadingDialog.show();
        HttpParams httpParams = new HttpParams();
        httpParams.put("3", "190125");
        httpParams.put("42", getMerNo());
        OkClient.getInstance().post(httpParams, new OkClient.EntityCallBack<BaseEntity>(context, BaseEntity.class) {
            @Override
            public void onError(Response<BaseEntity> response) {
                super.onError(response);
                loadingDialog.dismiss();

            }

            @Override
            public void onSuccess(Response<BaseEntity> response) {
                super.onSuccess(response);
                loadingDialog.dismiss();
                BaseEntity model = response.body();
                if (model == null) {
                    return;
                }

                if ("00".equals(model.getStr39())) {
                    List<WithdrawModel> list = JSONArray.parseArray(model.getStr57(), WithdrawModel.class);
                    mList.addAll(list);
                    mWithdrawListAdapter.setNewData(mList);
                }
            }
        });
    }

    @OnClick(R.id.iv_back)
    public void onViewClicked() {
        ViewUtils.overridePendingTransitionBack(context);
    }
}

File: app/src/main/java/com/kotyj/com/activity/mine/WithdrawListActivity.kt
package com.kotyj.com.activity.mine

import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.OrientationHelper
import com.alibaba.fastjson.JSONArray
import com.kotyj.com.R
import com.kotyj.com.adapter.WithdrawListAdapter
import com.kotyj.com.base.BaseActivity
import com.kotyj.com.model.BaseEntity
import com.kotyj.com.model.WithdrawModel
import com.kotyj.com.utils.okgo.OkClient
import com.lzy.okgo.model.HttpParams
import com.lzy.okgo.model.Response
import kotlinx.android.synthetic.main.general_mlist.*
import kotlinx.android.synthetic.main.layout_title.*
import java.util.*

class WithdrawListActivity : BaseActivity() {

    lateinit var mWithdrawListAdapter: WithdrawListAdapter
    private val mList = ArrayList<WithdrawModel>()


    override fun initLayout(): Int {
        return R.layout.general_mlist
    }

    override fun initData() {
        tv_title.text = "提现记录"
        recyclerView.addItemDecoration(DividerItemDecoration(context, OrientationHelper.VERTICAL))
        mWithdrawListAdapter = WithdrawListAdapter(mList)
        mWithdrawListAdapter.run {
            bindToRecyclerView(recyclerView)
            setEmptyView(R.layout.layout_bank_empty, recyclerView)
        }
        initListener()
        loadData()
    }

    private fun initListener() {
        iv_back.setOnClickListener { finish() }
        smartRefreshLayout.run {
            isEnableLoadmore = false
            setOnRefreshListener {
                loadData()
            }
        }
    }

    private fun loadData() {
        loadingDialog.show()
        val httpParams = HttpParams()
        httpParams.put("3", "190125")
        httpParams.put("42", merNo)
        OkClient.getInstance().post(httpParams, object : OkClient.EntityCallBack<BaseEntity>(context, BaseEntity::class.java) {
            override fun onError(response: Response<BaseEntity>) {
                super.onError(response)
                loadingDialog.dismiss()
                smartRefreshLayout.finishRefresh()
            }

            override fun onSuccess(response: Response<BaseEntity>) {
                super.onSuccess(response)
                loadingDialog.dismiss()
                smartRefreshLayout.finishRefresh()
                val model = response.body() ?: return

                if ("00" == model.str39) {
                    val list = JSONArray.parseArray(model.str57, WithdrawModel::class.java)
                    mList.addAll(list)
                    mWithdrawListAdapter!!.setNewData(mList)
                }
            }
        })
    }
}
